Scripts/Api.cs

```diff
@@ -123,25 +149,133 @@ public static IEnumerator CreateConversation(string token, Action<string> callba
             {
                 Logger.LogError($"Could not generate conversation; {e}");
                 throw;
-            }
+            }
+
+            // Need to dispose of WebRequest to prevent following error:
+            // "A Native Collection has not been disposed, resulting in a memory leak. Enable Full StackTraces to get more details."
+            request.Dispose();
         }
+
+        /// <summary>
+        /// Retrives the message history of an active conversation.
+        /// </summary>
+        /// <param name="token">Valid play-through token.</param>
+        /// <param name="conversationUuid">Active generated Conversation ID.</param>
+        /// <param name="minEventId">Minimum event Id from which to start reading message history. Only assign if you want messages after a certain point</param>
+        /// <param name="callback">Called when a valid history is generated.</param>
+        public static IEnumerator GetMessageHistory(string token, string conversationUuid, string minEventId, Action<GetMessageHistoryResponse> callback)
+        {
+            Dictionary<string, string> requestParams = new Dictionary<string, string>();
+            if(conversationUuid != default)
+            {
+                requestParams.Add("conversationUuid", conversationUuid);
+            }
+            if (minEventId != default)
+            {
+                requestParams.Add("minEventId", minEventId);
+            }
+
+            var request = UnityWebRequest.Get($"{BaseUrl}/play/message-history" + CreateQueryString(requestParams));
+            request.method = "GET"; 
+            request.SetRequestHeader("Authorization", $"Bearer {token}");
+
+            Logger.Log("Requesting message history...");
+
+            yield return request.SendWebRequest();
+
+            if (request.result != UnityWebRequest.Result.Success)
+            {
+                Logger.LogError("Error:" + request.error);
+                yield break;
+            }
+
+            var data = Encoding.UTF8.GetString(request.downloadHandler.data);
+
+            try
+            {
+                var deserialized = JsonConvert.DeserializeObject<GetMessageHistoryResponse>(data);
+                callback?.Invoke(deserialized);
+                Logger.Log("Conversation request complete");
+            }
+            catch (Exception e)
+            {
+                Logger.LogError($"Could not generate conversation; {e}");
```

Comment:
I think these logs are copy-pasted, would be worth writing custom messages (same for other endpoints)